nipype/interfaces/afni/preprocess.py

```diff
@@ -3475,29 +3476,38 @@ def _list_outputs(self):
         if not isdefined(self.inputs.out_file):
```

Comment:
All of this should not be here:

1. The `out_file` input spec can only define one of `name_source` or `genfile`. Since we are trying to deprecate `genfile`, `name_source` should be preferred.

2. In theory, `name_source` should fix this issue.

Reply:
@oesteban - i'm leaving the afni stuff to @Shotgunosine and others to figure out. right now, because of the way afni works with various kinds of extensions, the straight up namesource can run into issues.